thorlcr/activities/keyedjoin/thkeyedjoinslave.cpp

```diff
@@ -1408,21 +1433,22 @@ class CKeyedJoinSlave : public CSlaveActivity, implements IJoinProcessor, implem
                     partKeySet->addIndex(keyIndex.getClear());
                 }
                 keyManager.setown(createKeyMerger(helper->queryIndexRecordSize()->queryRecordAccessor(true), partKeySet, 0, nullptr, helper->hasNewSegmentMonitors(), false));
+                keyManagerCreated = true; // N.B. This isn't ensure previous line is not executed multiple times. For use in other methods, to confirm keyManager is ready to use.
```

Comment:
>N.B. This isn't ensure previous line is not executed multiple times.

that bit of the comment isn't correct, the keyManager pointer (line 1425) was/is sufficient for that - this is the only thread that will touch it. This boolean is only to prevent other threads (i.e. the thread that's collecting the stats) from accessing the keyManager before it's created.